lib/watir/elements/element.rb

```diff
@@ -493,6 +494,21 @@ def browser
       @query_scope.browser
     end
 
+    #
+    # Returns true if a previously located element is no longer attached to DOM.
+    #
+    # @return [Boolean]
+    # @see Watir::Wait
+    #
+
+    def stale?
+      raise Watir::Exception::Error, "Can not check staleness of unused element" unless @element
```

Comment:
`assert_exists` would attempt to relocate it and so long as that locator was in the DOM it would never show up as stale. I'm not sure a way to do this with our lazy loading other than:

```ruby
div = browser.div(id: 'foo')
div.exists?
browser.scroll.to(:end) until div.stale?
```